src/t_set.c

```diff
@@ -174,20 +174,29 @@ int setTypeAddAux(robj *set, char *str, size_t len, int64_t llval, int str_is_sd
                 return 1;
             }
         } else {
-            size_t maxelelen = intsetLen(set->ptr) == 0 ?
-                0 : max(sdigits10(intsetMax(set->ptr)),
-                        sdigits10(intsetMin(set->ptr)));
+            /* Check if listpack encoding is possible. */
```

Comment:
there's an over estimation here, which could mean that we won't use listpack for lists that would have fit.
i suppose it's better than violating lpSafeToAdd
```suggestion
            /* Check if listpack encoding is safe not to cross any threshold. */
```

Reply:
Yes, we need an upper bound, right?

This is a much smaller over estimation compared to the sdigits10 version though, where we used the length of the string representation for the estimate.